content/reference-strategy/index.md

```diff
@@ -0,0 +1,135 @@
+“key”.
+* Only able to reference one register per field.
+* Linking a full set of records is a special case.
+
+
+### CURIE
+
+A [CURIE](https://www.w3.org/TR/curie/) is a Compact URI so according to its
+definition for a given CURIE should be able to be transformed into its
+expanded URI form. The current use of CURIEs lacks any mechanism to resolve the
+prefix into its equivalent URL which means the expansion mechanism relies on
+the assumption that the prefix will be a known Register identifier, so to
+expand it you have to:
+
+1. Given a CURIE `country:GB`
+1. Take its prefix: `country`
+1. Compose a register subdomain: `country.register.gov.uk`
+1. Compose a record URI: `https://country.register.gov.uk/records/`
+1. Concatenate the URI with the CURIE value: `https://country.register.gov.uk/records/GB`
+
+In the case of a CURIE to the full set of records you don't need the last
+step:
+
+1. Given a CURIE `country:`
+1. Take its prefix: `country`
+1. Compose a register subdomain: `country.register.gov.uk`
+1. Compose a record URI: `https://country.register.gov.uk/records/`
+
+
+#### Benefits
+
+* Allows a field to reference a Record in any known register.
+* A data consumer can know the field contains references by the datatype of the
+  field.
+* It's built on a existing open standard. Technically it is a note and not a
+  recommendation but it is used in many standards, including RDF.
+
+#### Problems
+
+* A consumer has to know the assumptions that lead from a prefix to a URI. And
+  know it's not the standard way to work with CURIEs.
+* It's not possible to infer a type constraint for any new value.
+
+
+## Explanation
+
+Although Registers could be built on any of the exposed solutions, a CURIE has
+the edge in flexibility and builds on top of existing open standards for
+linking resources.
+
+### Expanding a CURIE
+
+The current solution for expanding CURIEs relies on a few assumptions. In
+order to remove this RFC proposes providing a **Context**, in a similar way
+[JSON-LD](https://json-ld.org/) does it.
+
+The definition of the context will happen at the Register Environment level,
+this way it can be guaranteed that a single context will work for any Register
+part of the Environment.
+
+A Context is a dictionary of prefixes and URIs, in JSON it would look like:
+
+```json
+{
+  "country": "https://country.register.gov.uk/records/",
+  "lae": "https://local-authority-eng.register.gov.uk/records/",
+  "example": "https://example.org/"
+}
+```
+
+With this context, the following expansions are expected to be true:
+
+* `country:GB` -> `https://country.register.gov.uk/records/GB`
+* `country:ZZ` -> `https://country.register.gov.uk/records/ZZ`
+* `lae:BIR` -> `https://local-authority-eng.register.gov.uk/records/BIR`
+* `example:32` -> `https://example.org/32`
+
+
+TODO: Where does the context live? It can't be part of the register API
+because a Register could be part of multiple environments.
```

Comment:
What is an example of a register being part of multiple environments?

Reply:
The immediate situation would be a Register being part of the register.gov.uk environment and being part of a local/test environment.

Another one, artificial case right now, would be a register from NHS that is part of the NHS environment and part of ours.

Reply:
Does that mean the CURIE doesn't resolve to a "single source of truth" but instead just one of many places where that truth is published?

Reply:
It resolves to the source of truth you decided. Which most of the time is our environment.

Reply:
@michaelabenyohai and I had a chat about another topic that led to a discussion around the possibility to encode more specific data instead of the CURIE (as per the spec) so it could be proved correct.  But right now it is just a thought.